storage.go

```diff
@@ -187,11 +188,27 @@ func (s *Storage) move(ctx context.Context, src string, dst string, opt pairStor
 	rs := s.getAbsPath(src)
 	rd := s.getAbsPath(dst)
 
-	// Create dir for dst path.
-	// Create dir before create file
-	err = os.MkdirAll(filepath.Dir(rd), 0755)
-	if err != nil {
-		return err
+	fi, err := os.Lstat(rd)
+	if err == nil {
+		// File is exist, let's check if the file is a dir.
+		// FIXME: maybe we need to handle symlink here?
```

Comment:
Do we need to return `services.ErrObjectModeInvalid` directly if dst is a symlink or return the error when the symlink target is a dir?

Reply:
LGTM